Detectors/TRD/base/src/CalDet.cxx

```diff
@@ -560,28 +543,25 @@ double TRDCalDet::calcMean(bool wghtPads)
   return calcMean(wghtPads, calib);
 }
 //_____________________________________________________________________________
-double TRDCalDet::calcRMS(bool wghtPads, int& calib)
+double CalDet::calcRMS(bool wghtPads, int& calib)
 {
   // Calculate the RMS value after rejection of the chambers not calibrated
   // wghPads = kTRUE weighted with the number of pads in case of a AliTRDCalPad created (t0)
   // calib = number of used chambers for the mean calculation
 
-  int iSM;
   double sum = 0.0;
   int ndet = 0;
   double meanALL = 0.0;
   double meanWP = 0.0;
-  double pads = 0.0;
   double padsALL = (144 * 16 * 24 + 144 * 12 * 6) * 18;
-  std::array<double, 18> meanSM{};
-  std::array<double, 18> meanSMWP{};
-  std::for_each(meanSM.begin(), meanSM.end(), [](double& n) { n = 0; });
-  std::for_each(meanSMWP.begin(), meanSMWP.end(), [](double& n) { n = 0; });
+  std::vector<double> meanSM(18);
+  std::vector<double> meanSMWP(18);
```

Comment:
Hi @bazinski,

I don't see the reason to use something beyond `std::array` for the definition of these quantities since they are fixed in size, thus generally faster. There are other places where I made this choice too, for example L74 in Detectors/TRD/base/include/TRDBase/CalDet.h.

I would agree to remove the two following lines `std::for_each` since they are not really necessary because array initializes all values to zero. Maybe `std::fill` would also ensure this too, but I am not sure about its performance compared to the `std::for_each` loop.

Reply:
Hi @jolopezl the reason for vector is that is zeros is internal data structure unlike array, besides that, if you dont cause it to expand its the same as an array performance wise.
Also fill is only optimised for char, the rest is a element wise walk through the structure.
So *besides* the boolean case, its faster to use memset to reset an array or vector.
There are examples in array??? of me using memset.
Maybe future versions of gcc and lvm will optimise the fill for basic types, but its not the case at the moment, just char.

Enjoy Bergen

Reply:
not precisely. array has its memory in place. vector will allocate somewhere else. array should be more cache friendly and you can serialize the objects much better.
You can init arrays using `std::array<double,10> a{}` to default init!

Reply:
No, we are not comparing `std::vector` with C-like arrays, but with `std::array` which is a template for a static size array, so it is initialized on the stack, while `std::vector` elements are allocated on the heap, causing an overhead.

As far as we have evolved in the code we have tried to optimize little things everywhere. So I guess there is no loss in doing the right choice by reverting just two lines.

Reply:
@sawenzel you mean stack vs heap? 
When I did a trivial test std::array does not intiialise its structure though.
mmmm I might go change a few things in other classes. I never bothered to check whether they ever *actually* change size after construction.

Reply:
@jolopezl sorry I replied while you were reply, ja I follow. Let me run along and convince myself ....

Reply:
`std::array<double,10> a{}`  **NOTE THE `{}`** will init.

Reply:
wow capitals mmmm let me change asap ....  ;-)